packages/xchain-ethereum/src/client.ts

```diff
@@ -79,10 +80,9 @@ export interface EthereumClient {
 
   isApproved(spender: Address, sender: Address, amount: BaseAmount): Promise<boolean>
   approve(
-    spender: Address,
-    sender: Address,
-    feeOptionKey?: FeeOptionKey,
-    amount?: BaseAmount,
+    params: ApproveParams & {
+      feeOptionKey?: FeeOptionKey
+    },
```

Comment:
Let's add `feeOptionKey` to `ApproveParams`. And for `estimateApprove` use `Omit<ApproveParams, 'feeOptionKey'>` as parameter.